pkg/tcpip/transport/udp/endpoint.go

```diff
@@ -201,7 +206,6 @@ func (e *endpoint) Close() {
 	e.net.Shutdown()
 	e.net.Close()
 	e.readShutdown = true
-	e.mu.Unlock()
 
 	e.waiterQueue.Notify(waiter.EventHUp | waiter.EventErr | waiter.ReadableEvents | waiter.WritableEvents)
```

Comment:
When calling in to `e.waiterQueue`, endpoint.go:Shutdown() holds `e.mu.Lock`. endpoint.go:HandlePacket() also holds `e.mu.Lock()` when called by forwarder.go:CreateEndpoint(). Only endpoint.go:OnICMPError() and endpoint.go:Close() did not.